sympy/geometry/ellipse.py

```diff
@@ -145,7 +145,10 @@ def __new__(
                 vradius = hradius * sqrt(1 - eccentricity**2)
 
         if hradius == vradius:
-            return Circle(center, hradius, **kwargs)
```

Comment:
If this fails to reduce to a Point, let's modify Circle to recognize this case instead of special casing it here.